crates/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -189,33 +189,53 @@ pub(super) fn datetime(
 fn apply_offsets_to_datetime(
     datetime: &Logical<DatetimeType, Int64Type>,
     offsets: &StringChunked,
-    offset_fn: fn(&Duration, i64, Option<&Tz>) -> PolarsResult<i64>,
     time_zone: Option<&Tz>,
 ) -> PolarsResult<Int64Chunked> {
```

Comment:
Just notice this now, but ideally the implementation should be in `polars-ops` or `polars-time` and then we only dispatch to those functions here. I'd like `polars-plan` to only be about the query planning/resolving.

Reply:
yup - ok if I do that separately?